audresample/core/lib.py

```diff
@@ -7,10 +7,21 @@
 root = os.path.dirname(os.path.realpath(__file__))
 
 bin_path = os.path.join(root, 'bin')
-if os.name == 'nt':
-    lib_path = os.path.join(bin_path, 'audresample.dll')  # pragma: no cover
+print(bin_path)
+print(os.path.exists(bin_path))
+print(os.path.exists(bin_path.replace('\\', '\\\\')))
+if os.name == 'nt':  # pragma: no cover
```

Comment:
did you try to check for `win32` as we do in `pyopensmile`? https://github.com/audeering/opensmile-python/blob/master/opensmile/core/SMILEapi.py#L12

Reply:
but ok, looking at your the commented print lines you already checked that it enters the `if` clause...

Reply:
maybe it's a problem of the `dll` itself and that it cannot resolve another dependency

Reply:
No, but I had the print statements at some point within the `if` statement and they produced output.

Reply:
But I'm also wondering why it is not working here, as with `opensmile-python` we have a working example.
Can it be related to the sox DLL that is also stored in the dir and needs to be loaded at some point?

Reply:
As far as I know - other than on Linux -  the directory where a `dll` is automatically add to the paths where it tries to resolve dependencies.

Reply:
But maybe there is another dependency missing.

Reply:
@agfcrespi how is the `soxr.dll` file included? I guess it is read by `audresample.dll`? And if yes, at what path it is expecting it?

Reply:
When I use Dependency Walker to inspect the dependencies of `audresample.dll` I get:

![image](https://user-images.githubusercontent.com/10383417/112469635-fd8eaa80-8d69-11eb-9321-1b0571a01c49.png)

so you can see that it finds `soxr.dll`, but there are also erros about other dependencies it cannot resolve. Not sure if this a hint...

Reply:
Sorry, I am really ignorant about Windows stuff, I don't have an answer for you at the moment. But I suspect we actually have some other problem before we get to the point of loading `soxr.dll`: we are seeing the same problem in `pyaudsp`, and we don't have other dependencies there (in addition to `libaudeering`). So maybe it's the same issue here.

Reply:
The missing files errors in Dependency Walker are normal and unrelated.

One reason why LoadLibrary might fail is when you try to load a 64-bit DLL into a 32-bit process, i.e. if your Python installation is for 32-bit (or the other way round).

Reply:
I installed audresample 0.1.4 on my local Windows and it works fine. Could this be an error in CI only, maybe due to a Python 32-bit installation?

Reply:
Maybe the Visual C++ Redistributable Runtime? As you can see in the screenshot you posted of Dependency Walker, it is linked dynamically to the runtime (VCRUNTIME140.dll) which means it has to be installed separately. We usually link statically to it so that it does not have this runtime dependency and this was done for opensmile but apparently not for audresamplelib.

Reply:
Very good catch!

Reply:
@agfcrespi could you check if that's the case and possibly rebuild the libraries by statically linking as @chausner-audeering suggested?

Reply:
How do you grab such a build?

I also wonder if https://github.com/audeering/audresamplelib should publish pre-build libraries as releases, so basically providing a ZIP file that you can download and get the library.

Reply:
`gitlab-ci.yml` does also not help much if we are on Github ;)

Reply:
Please forget what I wrote earlier. As we saw, the issue is with `soxr.dll` not `audresamplelib.dll`. It's an issue in our Conan wrapper for soxr, see internal link: https://gitlab.audeering.com/devops/conan/conan-soxr/-/issues/3

Reply:
I can provide a manually built binary for soxr.dll for now.

Reply:
Hmm, I ran into some other issues now, I guess needs more time to look into properly. @hagenw Please create a new release for now with macOS support.

Reply:
I looked further into this and it turns out that soxr also depends on vcomp140.dll which cannot be statically linked so it always has to be bundled. An alternative would be to build without OpenMP but this might make soxr slower so it's probably better to just bundle the library. I replaced soxr.dll with a version statically linked to the C++ runtime and also committed vcomp140.dll but it's still failing to load the library in the CI and I am not sure why.

Reply:
@hagenw and @chausner-audeering it seems all attempts have failed so far. You think we will be able to fix this any time soon? 

> An alternative would be to build without OpenMP but this might make soxr slower so it's probably better to just bundle the library. 

If this will fix the issue, we should do it. Since most our Python packages depend on `audresample` this issue is quite a blocker for Windows users.

Reply:
I am not sure when I'll have a chance to look further into it in the foreseeable future. At least the rest of this month, I am focused on other tasks already.

Reply:
>> An alternative would be to build without OpenMP but this might make soxr slower so it's probably better to just bundle the library.
> If this will fix the issue, we should do it. Since most our Python packages depend on audresample this issue is quite a blocker for Windows users.

I would vote for building without OpenMP for Windows then.

Reply:
I checked with [Dependencies](https://github.com/lucasg/Dependencies) and it seems that there are no missing dependencies:

![image](https://user-images.githubusercontent.com/10383417/122399850-73766c80-cf7b-11eb-92b7-d6f382e2fff6.png)

Reply:
I worked in my VM when I loaded the `soxr.dll` first:

```python
bin_path = os.path.join(root, 'bin')
if platform.system() == 'Windows':  # pragma: no cover
    # for some reason we have to load soxr.dll first...
    lib_path = os.path.join(bin_path, 'soxr.dll')
    ctypes.cdll.LoadLibrary(lib_path)
    lib_path = os.path.join(bin_path, 'audresample.dll')
...
lib = ctypes.cdll.LoadLibrary(lib_path)
```

No idea why. Let's see if it also works in the CI job...

Reply:
Great, you did it :)

Reply:
Yep, solves the issue. Weird :)

@hagenw @chausner-audeering if you have no objections I will merge and make a new release.

Reply:
Interesting workaround. I think it has to do with library search paths. By loading it explicitly beforehand, it will not need to be loaded indirectly when loading `audresample.dll` where maybe other search paths are used than when the script loads it directly. That would be my best guess what's happening.